File: apps/android/app/src/test/java/com/litter/android/ui/terminal/TerminalControlSequencesTests.kt
package com.litter.android.ui.terminal

import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Test

class TerminalControlSequencesTests {
    @Test
    fun `accessory controls include soft-keyboard terminal essentials`() {
        val labels = TerminalControlSequences.terminalAccessoryLabels

        listOf(
            "Missions",
            "Esc",
            "Tab",
            "Enter",
            "⌫",
            "Home",
            "End",
            "PgUp",
            "PgDn",
            "Ctrl-C",
            "Ctrl-D",
            "Ctrl-Z",
            "←",
            "↑",
            "↓",
            "→",
            "Paste",
            "Interrupt",
            "Close",
        ).forEach { label ->
            assertTrue("missing terminal accessory label $label", labels.contains(label))
        }
    }

    @Test
    fun `missions shortcut is terminal text not a native command token`() {
        assertEquals("/missions\n", TerminalControlSequences.DROID_MISSIONS_COMMAND)
    }

    @Test
    fun `navigation and control sequences match terminal byte conventions`() {
        assertEquals("\u001B", TerminalControlSequences.ESCAPE)
        assertEquals("\r", TerminalControlSequences.ENTER)
        assertEquals("\u007F", TerminalControlSequences.BACKSPACE)
        assertEquals("\u0003", TerminalControlSequences.CTRL_C)
        assertEquals("\u0004", TerminalControlSequences.CTRL_D)
        assertEquals("\u001B[5~", TerminalControlSequences.PAGE_UP)
        assertEquals("\u001B[6~", TerminalControlSequences.PAGE_DOWN)
    }
}

File: apps/ios/Tests/LitterTests/TerminalControlSequencesTests.swift
import XCTest
@testable import Litter

final class TerminalControlSequencesTests: XCTestCase {
    func testAccessoryControlsIncludeSoftKeyboardTerminalEssentials() {
        let labels = TerminalControlSequences.terminalAccessoryLabels

        [
            "Missions",
            "Esc",
            "Tab",
            "Enter",
            "⌫",
            "Home",
            "End",
            "PgUp",
            "PgDn",
            "Ctrl-C",
            "Ctrl-D",
            "Ctrl-Z",
            "←",
            "↑",
            "↓",
            "→",
            "Paste",
            "Interrupt",
            "Close"
        ].forEach { label in
            XCTAssertTrue(labels.contains(label), "missing terminal accessory label \(label)")
        }
    }

    func testMissionsShortcutIsTerminalTextNotNativeCommandToken() {
        XCTAssertEqual(TerminalControlSequences.droidMissionsCommand, "/missions\n")
    }

    func testNavigationAndControlSequencesMatchTerminalByteConventions() {
        XCTAssertEqual(TerminalControlSequences.escape, "\u{1B}")
        XCTAssertEqual(TerminalControlSequences.enter, "\r")
        XCTAssertEqual(TerminalControlSequences.backspace, "\u{7F}")
        XCTAssertEqual(TerminalControlSequences.ctrlC, "\u{03}")
        XCTAssertEqual(TerminalControlSequences.ctrlD, "\u{04}")
        XCTAssertEqual(TerminalControlSequences.pageUp, "\u{1B}[5~")
        XCTAssertEqual(TerminalControlSequences.pageDown, "\u{1B}[6~")
    }
}
